static/app/views/alerts/incidentRules/triggers/chart/thresholdsChart.tsx

```diff
@@ -79,28 +89,32 @@ export default class ThresholdsChart extends PureComponent<Props, State> {
     const {triggers, resolveThreshold} = this.props;
     const chartRef = this.ref?.getEchartsInstance?.();
     if (chartRef) {
-      this.updateChartAxis(
-        Math.max(
-          ...flatten(
-            triggers.map(trigger => [trigger.alertThreshold || 0, resolveThreshold || 0])
-          )
-        )
-      );
+      const thresholds = [
+        resolveThreshold || null,
+        ...triggers.map(t => t.alertThreshold || null),
+      ].filter(threshold => threshold !== null) as number[];
+      this.updateChartAxis(Math.min(...thresholds), Math.max(...thresholds));
     }
   };
 
   /**
    * Updates the chart so that yAxis is within bounds of our max value
    */
-  updateChartAxis = debounce((threshold: number) => {
-    const {maxValue} = this.props;
-    if (typeof maxValue !== 'undefined' && threshold > maxValue) {
-      // We need to force update after we set a new yAxis max because `convertToPixel`
-      // can return a negative position (probably because yAxisMax is not synced with chart yet)
-      this.setState({yAxisMax: Math.round(threshold * 1.1)}, this.forceUpdate);
-    } else {
-      this.setState({yAxisMax: null}, this.forceUpdate);
+  updateChartAxis = debounce((minThreshold: number, maxThreshold: number) => {
+    const {minValue, maxValue} = this.props;
+    let yAxisMax = maxValue ?? null;
+    let yAxisMin = minValue ?? null;
+
+    if (typeof maxValue !== 'undefined' && maxThreshold > maxValue) {
```

Comment:
```suggestion
    if (typeof maxValue === 'number' && maxThreshold > maxValue) {
```

prefer to confirm its the expected type

Reply:
good idea